Essentials/src/com/earth2me/essentials/Essentials.java

```diff
@@ -357,11 +373,28 @@ private void registerListeners(PluginManager pm) {
 
     @Override
     public void onDisable() {
+        boolean stopping = false;
+        if (isStopping != null) {
+            try {
+                stopping = (boolean) isStopping.invoke();
+            } catch (Throwable throwable) {
+                throwable.printStackTrace();
+            }
+        }
         for (User user : getOnlineUsers()) {
             if (user.isVanished()) {
                 user.setVanished(false);
                 user.sendMessage(tl("unvanishedReload"));
             }
+            if (stopping) {
+                user.setLogoutLocation();
+                if (!user.isHidden()) {
+                    user.setLastLogout(System.currentTimeMillis());
+                }
+                user.stopTransaction();
+                user.dispose();
```

Comment:
This is definitely the wrong thing to run here.

Should instead be user.cleanup() - user.dispose() will create an async task to eventually call user.cleanup, which will throw an exception. (Or should, since the server is shutting down).

Reply:
Ah okay, got confused on what did what :P